diff -Nur xpde.orig/xpde/src/core/xpde/uXPAPI_imp.pas xpde.chng/xpde/src/core/xpde/uXPAPI_imp.pas
--- xpde.orig/xpde/src/core/xpde/uXPAPI_imp.pas	Mon Apr  7 15:01:22 2003
+++ xpde.chng/xpde/src/core/xpde/uXPAPI_imp.pas	Tue May  6 11:58:57 2003
@@ -179,7 +179,7 @@
 
 function TXPAPI.GetAppDir: string;
 begin
-    result:=BaseDir+'bin/';
+    result:='/usr/bin/';
 end;
 
 function TXPAPI.GetBitmapsDir: string;
@@ -475,43 +475,18 @@
 end;
 
 function TXPAPI.GetBaseDir: string;
-var
-    reg: TRegistry;
-    apdir: string;
-    i: integer;
-    f: integer;
 begin
-    reg:=TRegistry.create;
-    try
-        if reg.OpenKey('Software/XPde/System',false) then begin
-            result:=reg.ReadString('basedir');
-        end
-        else begin
-            f:=0;
-            apdir:=extractfilepath(application.exename);
-            for i:=length(apdir) downto 1 do begin
-                if apdir[i]='/' then begin
-                    inc(f);
-                    if (f=3) then begin
-                        result:=copy(apdir,1,i);
-                        break;
-                    end;
-                end;
-            end;
-        end;
-    finally
-        reg.free;
-    end;
+    result := '/usr/share/xpde/';
 end;
 
 function TXPAPI.GetAppletsDir: string;
 begin
-    result:=Appdir+'applets/';
+    result:=Appdir;
 end;
 
 function TXPAPI.GetAppsDir: string;
 begin
-    result:=Appdir+'apps/';
+    result:=Appdir;
 end;
 
 function TXPAPI.ReplaceSystemPaths(const path: string): string;
